sky/backends/cloud_vm_ray_backend.py

```
@@ -2374,29 +2374,31 @@ def _internal_ips(self) -> Optional[List[str]]:
             return [ips[0] for ips in self.stable_internal_external_ips]
         return None
 
-    def internal_ips(
-            self,
-            max_attempts: int = _FETCH_IP_MAX_ATTEMPTS) -> Optional[List[str]]:
+    def internal_ips(self,
+                     max_attempts: int = _FETCH_IP_MAX_ATTEMPTS) -> List[str]:
         internal_ips = self._internal_ips
         if internal_ips is not None:
             return internal_ips
         self.update_cluster_ips(max_attempts=max_attempts)
-        return self._internal_ips
+        internal_ips = self._internal_ips
+        assert internal_ips is not None, 'update_cluster_ips failed.'
```

Shall we raise an error here?